primitives/avn-common/src/event_discovery.rs

```diff
@@ -0,0 +1,198 @@
+use crate::*;
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use event_types::EthEvent;
+use sp_core::{bounded::BoundedBTreeSet, ConstU32};
+use sp_runtime::traits::Saturating;
+
+use self::event_types::ValidEvents;
+
+pub const MAX_INCOMING_EVENTS_BATCH_SIZE: u32 = 32u32;
+pub type IncomingEventsBatchLimit = ConstU32<MAX_INCOMING_EVENTS_BATCH_SIZE>;
+
+#[derive(
+    Encode, Decode, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Default, TypeInfo, MaxEncodedLen,
+)]
+pub struct EthBlockRange {
+    pub start_block: u32,
+    pub length: u32,
+}
+
+impl EthBlockRange {
+    pub fn next_range(&self) -> EthBlockRange {
+        EthBlockRange {
+            start_block: self.start_block.saturating_add(self.length),
+            length: self.length,
+        }
+    }
+    pub fn range(&self) -> (u32, u32) {
+        let end_block = self.start_block.saturating_add(self.length).saturating_less_one();
```

Comment:
Why do we remove 1? Not sure if the tests mirror this.

Reply:
EthBlockRange should be `[start_block, end_block)` where `end_block = start_block + length`

Reply:
Exactly but if am reading the code correctly, we are doing `end_block = start_block + length - 1`

Reply:
so the range is exclusive of the last block 
so for an instance of EthereumRange { start:block: 0 , length: 100 }

range 1 is : mathematically `[0,100)` or `[0,99]`. Rust wise is `0..=99` or `0..100`
range 2 is : mathematically `[100,200)` or `[100,199]`. Rust wise is `100..=199` or `100..200`

I think the formula does that

Reply:
But since we are talking about this its worth validating that when we query the range returned in the client, the full range (including the range start and end)